### DISCUSSION TITLE
Trying to use "SFTP as storage backend"

### DISCUSSION THREAD

**Discussion by JanneHarju** [drakkan/sftpgo#753] [Q&A] [answered]
I have another sftp server which I need to connect in sftpgo. So I tried to connect my other sftp-server using FileZilla and it works fine. When I try same settings in sftpgo it fails. So if I connect sftpgo-Folder, which is connected to other sftp, to sftpgo-User then I cannot connect to sftpgo using that sftpgo-User. In log I found this line:
`{"level":"warn","time":"2022-03-09T09:33:32.352","sender":"SSH","connection_id":"c8f6b4fa449cc2b23824bdc6e441bd54e619f2d377088bc17a2951e054ee609b","message":"could not accept a channel: EOF"}`
What could be reason for this?

**Comment by drakkan** [comment, marked as answer]
Hi,

the log above seems unrelated, please post more logs and the configuration for the SFTP backend, individual settings or a screen from the web UI. Thanks

EDIT: this was a firewall issue, the SFTPGo container was unable to connect to the configured SFTP server

**Reply by JanneHarju** [reply]
I mark this as answer even it is not to get this issue as green.

**Comment by JanneHarju** [comment]
![image](https://user-images.githubusercontent.com/17338490/157426698-ef31f1f3-25bb-422e-a493-c8ad34d2e086.png)
![image](https://user-images.githubusercontent.com/17338490/157426949-d3460648-fdfd-4449-9b4f-008ce0d01b67.png)
I tried to remove all customer related data. Hopefully this still show you how configured. In user picture there is two virtual folders. One is using folder which is connected to Azure File Storage and other is connected to other SFTP-server.
There is lot of some other customers in same sftpgo server so log is quite hard to read when there is other communications as well. Should logs be after I try to connect to sftpgo using Filezilla or after I configure sftpgo-Folder?

**Reply by drakkan** [reply]
> ![image](https://user-images.githubusercontent.com/17338490/157426698-ef31f1f3-25bb-422e-a493-c8ad34d2e086.png) ![image](https://user-images.githubusercontent.com/17338490/157426949-d3460648-fdfd-4449-9b4f-008ce0d01b67.png) I tried to remove all customer related data. Hopefully this still show you how configured. In user picture there is two virtual folders. One is using folder which is connected to Azure File Storage and other is connected to other SFTP-server. There is lot of some other customers in same sftpgo server so log is quite hard to read when there is other communications as well. Should logs be after I try to connect to sftpgo using Filezilla or after I configure sftpgo-Folder?

The configuration looks good, please post the logs after you connect with filezilla, you can filter by connection_id

**Reply by JanneHarju** [reply]
`{"level":"debug","time":"2022-03-09T10:57:15.017","sender":"dataprovider_postgresql","connection_id":"14d7f633d7e8b745347e36a624fb90d96dfc01999481bc539d5952fd7412d011","message":"create intermediary dir to \"/My_company1\", path \"/srv/sftpgo/data/My_company1ERPUser/My_company1\", err: <nil>"}
`
with same connection id first there is this and then 
`{"level":"debug","time":"2022-03-09T10:59:28.552","sender":"dataprovider_postgresql","connection_id":"14d7f633d7e8b745347e36a624fb90d96dfc01999481bc539d5952fd7412d011","message":"create intermediary dir to \"/My_company1_unifaun\", path \"/srv/sftpgo/data/My_company1ERPUser/My_company1_unifaun\", err: <nil>"}
{"level":"info","time":"2022-03-09T10:59:28.552","sender":"SSH","connection_id":"14d7f633d7e8b745347e36a624fb90d96dfc01999481bc539d5952fd7412d011","message":"User \"My_company1ERPUser\" logged in with \"password\", from ip \"12.34.56.78\", client version \"SSH-2.0-FileZilla_3.55.1\""}
{"level":"debug","time":"2022-03-09T10:59:28.552","sender":"common","connection_id":"14d7f633d7e8b745347e36a624fb90d96dfc01999481bc539d5952fd7412d011","message":"ssh connection added, num open connections: 1"}
{"level":"warn","time":"2022-03-09T10:59:28.552","sender":"SSH","connection_id":"14d7f633d7e8b745347e36a624fb90d96dfc01999481bc539d5952fd7412d011","message":"could not accept a channel: EOF"}
{"level":"debug","time":"2022-03-09T10:59:28.552","sender":"common","connection_id":"14d7f633d7e8b745347e36a624fb90d96dfc01999481bc539d5952fd7412d011","message":"ssh connection removed, num open ssh connections: 0"}`
I found this log by searching filezilla. I noticed that client name and version is logged. So I assume this is my FileZilla connection.

**Reply by JanneHarju** [reply]
Sorry about bad log format.

**Reply by JanneHarju** [reply]
And if it might affect to something, I'm using docker in Azure container instance.

**Reply by JanneHarju** [reply]
And if I remove SFTP related virtual folder from that user connection works as it should be. So it got broken if I use that sftp-folder with that user.

**Reply by JanneHarju** [reply]
Sorry for spam but one thing what could tell something. I have two separete sftpgo-servers. If I configure one sftpgo to use another sftpgo-server as SFTP-server it works as intended when I connect to first one. But this is only my test scenario not final one which need to be used with other sftp-esrver.

**Reply by drakkan** [reply]
I'm unable to see any useful info in the above logs sorry, you should at least have a log like this one

```
login without host key validation, please provide at least a fingerprint!
```

when sftpgo logins to your sftp backend since you don't provided the fingerprint. 

Is the remote server OpenSSH or another SFTP server?

**Reply by JanneHarju** [reply]
For what I know it is sftp server. Or so it has been spoken to me. I will check that fingerprint part.

**Reply by JanneHarju** [reply]
Hmm. After I restart sftpgo FileZilla ask something about fingerprint when connecting first time.

**Reply by drakkan** [reply]
> For what I know it is sftp server. Or so it has been spoken to me. I will check that fingerprint part.

There are a lot of SFTP servers out of there, some require special configurations. Try to set a buffer size just in case this is something like aws sftp transfer, but without a proper log this is just a guess

**Reply by JanneHarju** [reply]
If it is aws sftp what would be good buffer size, because I have no idea should it be 1 or 1000.

**Reply by drakkan** [reply]
> If it is aws sftp what would be good buffer size, because I have no idea should it be 1 or 1000.

1 is enough

**Reply by JanneHarju** [reply]
I tried 1 and no luck so far.
This is what FileZilla log says after trying to connect:

Status:	Connecting to mycompany.northeurope.cloudapp.azure.com:2022...
Status:	Using username "MyUserName". 
Command:	Pass: *********
Error:	Connection timed out after 20 seconds of inactivity
Error:	Could not connect to server
Status:	Waiting to retry...
Status:	Connecting to mycompany.northeurope.cloudapp.azure.com:2022...
Response:	fzSftp started, protocol_version=11

And for yout earlier comment about fingerprint. There was 0 logs about fingerprint.
`login without host key validation, please provide at least a fingerprint!`

**Reply by drakkan** [reply]
> I tried 1 and no luck so far. This is what FileZilla log says after trying to connect:
> 
> Status: Connecting to mycompany.northeurope.cloudapp.azure.com:2022... Status: Using username "MyUserName". Command: Pass: ********* Error: Connection timed out after 20 seconds of inactivity Error: Could not connect to server Status: Waiting to retry... Status: Connecting to mycompany.northeurope.cloudapp.azure.com:2022... Response: fzSftp started, protocol_version=11
> 
> And for yout earlier comment about fingerprint. There was 0 logs about fingerprint. `login without host key validation, please provide at least a fingerprint!`

So basically SFTPGo seems unable to connect to the configured SFTP server. If you are sure that the SFTP server address and credentials are correct and that the SFTP server is reachable from the SFTPGo container, I have no other ideas sorry. 
If you can send some test credentials for your SFTP server to my email address, I can try it myself and investigate further, but this is a service I offer to project [sponsors](https://github.com/sponsors/drakkan)

**Reply by JanneHarju** [reply]
Thanks for your support. I have only asked from admins to configure to firewall spesific IP(public ip for my azure container instance) with spesific port. I'm not sure how to test connection inside docker container. In Azure container instance there is posibility to connect using bash. but it seems that there is no much tools to test connection. 
![image](https://user-images.githubusercontent.com/17338490/157440432-449e43b6-ed40-4563-a22e-10ecf162a488.png)
But I'm afraid that problem is not with my configuration but in some network/firewall configuration which I cannot check/do. But this was helpfull to know that is should try to find problem from somewhere else.

**Reply by JanneHarju** [reply]
It was firewall issue after all. It have different IP in outbound traffic than which IP you need to connect to it. Azure thing.

**Reply by drakkan** [reply]
> It was firewall issue after all. It have different IP in outbound traffic than which IP you need to connect to it. Azure thing.

Thanks for reporting back, I edited the selected answer
